File: src/QuizWebApi/Models/Admin/QuizDefinition.cs
using System;
using System.Collections.Generic;

namespace QuizWebApi.Models.Admin
{
    /// <summary>
    /// 
    /// </summary>
    public class QuizDefinition : Status
    {
        public QuizDefinition()
        {
            SponsorList = new List<SponsorDetail>();
        }
        public string DocumentType { get; set; }
        public string QuizName { get; set; }
        public string QuizDomainHost { get; set; }
        public string QuizType { get; set; }
        public int NoOfQuestions { get; set; }
        public int NoOfParticipants { get; set; }
        public int QuizDurationTime { get; set; }
        public string QuizDurationType { get; set; }
        public DateTime QuizStartTime { get; set; }
        public DateTime QuizEndTime { get; set; }
        public bool ShuffleQuestions { get; set; }
        public bool IsQuizFromLargerPool { get; set; }
        public int NoOfQuestionsInPool { get; set; }
        public bool AllowConcurrentAccess { get; set; }
        public string ParticipantType { get; set; }
        public bool IsQuizAutoEvaluate { get; set; }
        public bool ShowScoreAfterAttempt { get; set; }
        public bool PostScoreOnSocialMedia { get; set; }
        public string Status { get; set; }
        public string Stage { get; set; }
        public string MessageBeforeQuizTime { get; set; }
        public string MessageAfterQuizTime { get; set; }
        public RegistrationFields RegistrationFields { get; set; }
        public List<SponsorDetail> SponsorList { get; set; }
        public string CreatedBy { get; set; }
        public DateTime CreatedOn { get; set; }
        public DateTime ModifiedOn { get; set; }
    }
}

File: src/QuizUI/src/app/models/QuizDefinition.ts
export class FormData {
  QuizName: string = '';
  QuizDomainHost: string = '';
  QuizType: string = '';
  NoOfQuestions: number;
  NoOfParticipants: number;
  QuizDurationTime: number;
  QuizDurationType: string;
  QuizStartTime: any;
  QuizEndTime: any;
  ShuffleQuestions: boolean;
  IsQuizFromLargerPool: boolean;
  NoOfQuestionsInPool: number;
  AllowConcurrentAccess: boolean;
  ParticipantType: string = '';
  IsQuizAutoEvaluate: boolean;
  ShowScoreAfterAttempt: boolean;
  PostScoreOnSocialMedia: boolean;
  Status: string = '';
  Stage: string = '';
  MessageBeforeQuizTime: string = '';
  MessageAfterQuizTime: string = '';
  QuestionNo: number;
  QuestionText: string = '';
  IsImageneeded: boolean;
  ImageUrl: string = '';
  AnswerType: string = '';
  Answer: string = '';
  Score: string = '';
  Option1: string = '';
  Option2: string = '';
  Option3: string = '';
  Option4: string = '';
  IsTeamName: boolean;
  IsEmail: boolean;
  IsValidateEmail: boolean;
  IsContestantName: boolean;
  IsPhone: boolean;
  IsContact: boolean;
  RulesAndRegulations: string = '';
  RegistrationFields = new RegistrationFields();
  SponsorList = new Array<SponsorDetail>();
  Questions = new Array<QuizSet>();
  IsEditQuestion: boolean;

  clear() {
    this.QuizName = '';
    this.QuizDomainHost = '';
    this.QuizType = '';
    this.NoOfQuestions = 0;
    this.NoOfParticipants = 0;
    this.QuizDurationTime = 0;
    this.QuizDurationType = '';
    this.QuizStartTime = '';
    this.QuizEndTime = '';
    this.ShuffleQuestions = true;
    this.IsQuizFromLargerPool = true;
    this.NoOfQuestionsInPool = 0;
    this.AllowConcurrentAccess = true;
    this.ParticipantType = '';
    this.IsQuizAutoEvaluate = true;
    this.ShowScoreAfterAttempt = true;
    this.PostScoreOnSocialMedia = true;
    this.Status = '';
    this.Stage = '';
    this.MessageBeforeQuizTime = '';
    this.MessageAfterQuizTime = '';
    this.QuestionNo = 0;
    this.QuestionText = '';
    this.IsImageneeded = false;
    this.ImageUrl = '';
    this.AnswerType = '';
    this.Answer = '';
    this.Score = '';
    this.Option1 = '';
    this.Option2 = '';
    this.Option3 = '';
    this.Option4 = '';
    this.IsTeamName = true;
    this.IsEmail = true;
    this.IsValidateEmail = true;
    this.IsContestantName = true;
    this.IsPhone = true;
    this.IsContact = true;
    this.RulesAndRegulations = '';
    this.RegistrationFields = new RegistrationFields();
    this.SponsorList = new Array<SponsorDetail>();
    this.Questions = new Array<QuizSet>();
    this.IsEditQuestion = false;
  }
}

export class QuizDefinition {
  QuizName: string;
  QuizDomainHost: string;
  QuizType: string;
  NoOfQuestions: number;
  NoOfParticipants: number;
  QuizDurationTime: number;
  QuizDurationType: string;
  QuizStartTime: any;
  QuizEndTime: any;
  ShuffleQuestions: boolean;
  IsQuizFromLargerPool: boolean;
  NoOfQuestionsInPool: number;
  AllowConcurrentAccess: boolean;
  ParticipantType: string;
  IsQuizAutoEvaluate: boolean;
  ShowScoreAfterAttempt: boolean;
  PostScoreOnSocialMedia: boolean;
  Status: string;
  Stage: string;
  MessageBeforeQuizTime: string;
  MessageAfterQuizTime: string;
  RegistrationFields = new RegistrationFields();
  SponsorList = new Array<SponsorDetail>();
}

export class RegistrationFields {
  IsTeamName: boolean;
  IsEmail: boolean;
  IsValidateEmail: boolean;
  IsContestantName: boolean;
  IsPhone: boolean;
  IsContact: boolean;
  RulesAndRegulations: string;
}

export class SponsorDetail {
  Path: string;
  Position: string;
}

export class QuizQuestions {
  QuizName: string;
  QuizType: string;
  Questions = new Array<QuizSet>();
}

export class QuizSet {
  QuestionNo: number;
  QuestionText: string;
  IsImageneeded: boolean;
  ImageUrl: string;
  AnswerType: string;
  Answer: string;
  Score: string;
  Option1: string;
  Option2: string;
  Option3: string;
  Option4: string;
}
